### PR TITLE
Feature/proxy 862

### PR DESCRIPTION
**Pull request by alex-sandrk**
Example of release: https://github.com/alex-sandrk/WalletDesktop/releases/tag/1.0.54

https://github.com/Lumerin-protocol/WalletCore/pull/12

PR should be merged after Github environments are configured

### CODE REVIEW

File: .devGoerli.env [hunk 1]
```diff
@@ -0,0 +1,19 @@
+DISPLAY_NAME=Goerli
+CHAIN_ID=5
+SYMBOL=ETH
+
+LUMERIN_TOKEN_ADDRESS=0xF3aCe2847F01D3ef1025c7070579611091A6422D
+CLONE_FACTORY_ADDRESS=0xbF2A6EA18e2CF0846cE7FC9Fa9EB9bA22BF035fF
+
+PROXY_ROUTER_URL=proxyrouter.stg.lumerin.io:8080
```
Comment (lsheva) [inline .devGoerli.env:8]: protocol is missing

**Review by lsheva — APPROVED** [review]
Neat!

File: .devGoerli.env (hunk 1 above)
Comment (lsheva) [reply .devGoerli.env:8]: ignore it, let's keep as it is

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -12,19 +12,62 @@ jobs:
...
+      - name: Set env from gitlab
+        run: |
+          curl --header "PRIVATE-TOKEN: ${{ secrets.GITLAB_GROUP_TOKEN }}" "https://gitlab.com/api/v4/groups/${{ secrets.GITLAB_GROUP_ID }}/variables" | jq -c '.[]' | 
+            while IFS= read -r line; do
+              scope=$(<<<"$line" jq -r .environment_scope);
+              if [ "${scope}" = "${{ matrix.environment }}" ] || [ "${scope}" = '*' ]
+              then
+                key=$(<<<"$line" jq -r .key);
+                value=$(<<<"$line" jq -r .value);
+                echo "$key=$value" >> .env
```
Comment (lsheva) [inline .github/workflows/auto-release.yml:67]: just realized that the bundle will include all of the secrets defined on the group level in GITLAB
we should think about some other way

File: .devGoerli.env (hunk 1 above)
Comment (alex-sandrk) [reply .devGoerli.env:8]: discussed in the meeting: we will use URL with protocol

File: .github/workflows/auto-release.yml (hunk 2 above)
Comment (alex-sandrk) [reply .github/workflows/auto-release.yml]: added filtering to use only specified variables

File: .devGoerli.env [hunk 3]
```diff
@@ -0,0 +1,19 @@
+DISPLAY_NAME=Goerli
+CHAIN_ID=5
+SYMBOL=ETH
+
+LUMERIN_TOKEN_ADDRESS=0xF3aCe2847F01D3ef1025c7070579611091A6422D
+CLONE_FACTORY_ADDRESS=0xbF2A6EA18e2CF0846cE7FC9Fa9EB9bA22BF035fF
+
+PROXY_ROUTER_URL=https://proxyrouter.stg.lumerin.io:8080
```
Comment (lsheva) [inline .devGoerli.env:8]: http://proxy.dev.lumerin.io:8080/

File: src/components/devices/Device.js [hunk 4]
```diff
@@ -100,7 +100,7 @@ const Device = ({
   isLoading,
   status = 'Miner status unavailable',
   isPrivilegedApiAvailable,
-  proxyRouterUrl,
+  proxyRouterPool,
```
Comment (srt0422) [inline src/components/devices/Device.js:103]: I don't think this name change is correct.

Comment (srt0422) [reply src/components/devices/Device.js:103]:
```suggestion
  proxyRouterUrl,
```

File: src/components/devices/Device.js [hunk 5]
```diff
@@ -132,7 +132,7 @@ const Device = ({
       </dl>
       <div className="pool-row row">{poolAddress}</div>
       <div className="pool-user-row row">{poolUser}</div>
-      {poolAddress !== proxyRouterUrl && !isLoading && (
+      {poolAddress !== proxyRouterPool && !isLoading && (
```
Comment (srt0422) [inline src/components/devices/Device.js:135]:
```suggestion
      {poolAddress !== proxyRouterUrl && !isLoading && (
```

File: src/components/devices/Devices.js [hunk 6]
```diff
@@ -51,6 +51,10 @@ const ConfigureBtn = styled(Btn)`
 `;
 
 const Devices = props => {
+  const STRATUM_PROXY_ROUTER_PORT = 3333;
+  const proxyRouterHost = new URL(props.proxyRouterUrl).hostname;
+  const proxyRouterPool = `stratum+tcp://${proxyRouterHost}:${STRATUM_PROXY_ROUTER_PORT}`;
```
Comment (srt0422) [inline src/components/devices/Devices.js:56]: I think we just need to call it proxyRouterUrl.  that's what it is.  not a pool.

**Review by lsheva — APPROVED** [review]

**Pull request merged**